service/service.go

```diff
@@ -430,6 +437,29 @@ func (s *service) BeforeServe(
 		} else {
 			log.Println("No management servers found.")
 		}
+
+		// Access the storagearrays key (which is a slice of maps)
+		storageArrays := secretParams.Get("storagearrays").([]interface{})
```

Comment:
Won't this panic if "storagearrays" is missing in the secret?

Reply:
seems like it, thanks for that. will fix.